### PR TITLE
TVB-2913: Separate the storage handling in a different module PART 2

### PR DESCRIPTION
**Pull request by VinczeRobert**
The branch from this PR was made from the previous one and was currently rebased.
I left the WIP tag, altough I am not sure if there are still things to do besides testing.
There is one problem though related to a path_for call from the REST client and I have not managed to find a good solution for it yet.

### CODE REVIEW

File: framework_tvb/tvb/core/neocom/h5.py [hunk 1]
```diff
@@ -139,6 +143,22 @@ def store_complete(datatype, base_dir, generic_attributes=GenericAttributes()):
...
 
+def store_complete_to_dir(datatype, base_dir, generic_attributes=GenericAttributes()):
+    h5_class = REGISTRY.get_h5file_for_datatype(datatype.__class__)
+    storage_path = path_by_dir(base_dir, h5_class, datatype.gid)
+
+    index_inst = store_complete(datatype, storage_path, h5_class, generic_attributes)
+    return index_inst
+
+
+def store_complete_to_op_dir(datatype, op_id, project_name, generic_attributes=GenericAttributes()):
```
Comment (popaula937) [inline framework_tvb/tvb/core/neocom/h5.py:154]: Could this remain called `store_complete` and the common `store_complete` function at line 128 be private?

File: framework_tvb/tvb/core/entities/file/file_update_scripts/005_update_files.py [hunk 2]
```diff
@@ -150,6 +151,16 @@ def _migrate_connectivity(**kwargs):
         hemispheres = int(right_regions) * [True] + int(left_regions) * [False]
         storage_manager.store_data(numpy.array(hemispheres), 'hemispheres')
 
+    try:
+        storage_manager.get_data('hemispheres')
+    except MissingDataSetException:
```
[the remark this answers is not in the document]
Comment (VinczeRobert) [reply framework_tvb/tvb/core/entities/file/file_update_scripts/005_update_files.py]: I accidentally included a commit from you, but now I eliminated it from this PR.

File: framework_tvb/tvb/interfaces/web/controllers/hpc_controller.py [hunk 3]
```diff
@@ -6,7 +6,7 @@
 # TheVirtualBrain-Scientific Package (for simulators). See content of the
 # documentation-folder for more details. See also http://www.thevirtualbrain.org
 #
-# (c) 2012-2022, Baycrest Centre for Geriatric Care ("Baycrest") and others
+# (c) 2012-2020, Baycrest Centre for Geriatric Care ("Baycrest") and others
```
[the remark this answers is not in the document]
Comment (VinczeRobert) [reply framework_tvb/tvb/interfaces/web/controllers/hpc_controller.py:9]: I will do some UI testing for this PR as well to see if there are more wrongly merged parts that are not detected by our tests.

File: framework_tvb/tvb/core/services/project_service.py [hunk 4]
```diff
@@ -459,6 +460,10 @@ def get_project_structure(self, project, visibility_filter, first_level, second_
             data = {}
             is_group = False
             group_op = None
+            dt_entity = dao.get_datatype_by_gid(dt.gid)
+            if dt_entity is None:
+                self.logger.warning("Ignored entity (possibly removed DT class)" + str(dt))
+                continue
```
[the remark this answers is not in the document]
Comment (VinczeRobert) [reply framework_tvb/tvb/core/services/project_service.py:466]: Probably another merging mistake.

File: tvb_storage/tvb/storage/storage_interface.py [hunk 5]
```diff
@@ -63,6 +63,9 @@ class StorageInterface:
 
     ZIP_FILE_EXTENSION = "zip"
 
+    H5_EXTENSION = '.h5'
+    H5_FILE_NAME_STRUCTURE = '{}_{}.h5'
```
[the remark this answers is not in the document]
Comment (VinczeRobert) [reply tvb_storage/tvb/storage/storage_interface.py:67]: You are talking only about their name, right? Like "FILE_EXTENSION" and "FILE_NAME_STRUCTURE"?

[the remark this answers is not in the document]
Comment (popaula937) [reply tvb_storage/tvb/storage/storage_interface.py:67]: yes

File: framework_tvb/tvb/core/neocom/h5.py (hunk 1 above)
Comment (VinczeRobert) [reply framework_tvb/tvb/core/neocom/h5.py]: Of course

**Review by popaula937 — APPROVED** [review]
As long as this is well tested, I agree with the changes, just make sure that new code from master does not use the old form of these changed methods.

**Pull request merged**